### DISCUSSION TITLE
Customizing Welcome Page

### DISCUSSION THREAD

**Discussion by ntrouve-onera** [open-webui/open-webui#3053] [Q&A]
Hi, first awesome work and thanks!

I'm looking into a way to customize the experience (especially the front page) where it currently says "Welcome XXX, How can I help you today". 

Ideally I would like to display pictures, some text, maybe a couple html links. Also edit the "LLM can make mistakes" section to add specific warning adapted to my security context.

From looking in the source I can't even find where that page is set up. I only find those strings in the translation/localization directory (src/lib/il8n/locales/xx-XX/ folders).

Anyone can help me find where to work?

Sorry if It was already adressed elsewhere I couldn't find info in the documentation nor here.

**Comment by JuanjoJG-WATA** [comment]
Yep, you need to modify some .svelte files from **open-webui/src/lib/components/chat/**

The one with the "LLM can make mistakes" is called **MessageInput.svelte.**, for the rest you'll need to explore a bit, as I have never tweaked with them.

To commit the changes after you are done editing the files (I use VSC, as the Search tab is really useful), you need to change directories to open-webui again and do `npm run build`. After a while the changes will be done.

Have fun!

**Comment by ntrouve-onera** [comment]
Ok, thanks indeed *.svelte were not seen as text files.

Regarding npm build, I didn't install using that manual installation as I couldn't get the packages to download through my proxy/micropip/node.js. How would that work with the pip package install method?

**Reply by JuanjoJG-WATA** [reply]
Oh shoot. Maybe try this, but I'm not 100% sure it will work, as I installed cloning the repo and using Node.js and Python.

1. run "npm run dev" instead of "npm run build"
2. run the backend with the bash script or "uvicorn main:app --host 0.0.0.0 --port {YOUR_PORT} --forwarded-allow-ips '*' --reload"
3. now access the front-end using the dev url/port (default is localhost:5173)
